packages/mui-lab/src/internal/pickers/hooks/usePickerState.ts

```diff
@@ -95,6 +102,9 @@ export function usePickerState<TInput, TDateValue>(
       onClear: () => acceptDate(valueManager.emptyValue, true),
       onAccept: () => acceptDate(draftState.draft, true),
       onDismiss: () => setIsOpen(false),
+      onCancel: () => {
+        acceptDate(initialDate as unknown as TDateValue, true);
```

Comment:
Can we set the type of `initialDate` to `TDateValue` when declaring it rather than doing it here?